builtin/k8s/platform.go

```diff
@@ -263,37 +263,48 @@ func (p *Platform) Deploy(
 		}
 	}
 
-	// Update the deployment with our spec
-	deployment.Spec.Template.Spec = corev1.PodSpec{
-		Containers: []corev1.Container{
-			{
-				Name:            result.Name,
-				Image:           img.Name(),
-				ImagePullPolicy: pullPolicy,
-				Ports:           containerPorts,
-				LivenessProbe: &corev1.Probe{
-					Handler: corev1.Handler{
-						TCPSocket: &corev1.TCPSocketAction{
-							Port: intstr.FromInt(defaultPort),
-						},
-					},
-					InitialDelaySeconds: initialDelaySeconds,
-					TimeoutSeconds:      timeoutSeconds,
-					FailureThreshold:    failureThreshold,
+	container := corev1.Container{
+		Name:            result.Name,
+		Image:           img.Name(),
+		ImagePullPolicy: pullPolicy,
+		Ports:           containerPorts,
+		LivenessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				TCPSocket: &corev1.TCPSocketAction{
+					Port: intstr.FromInt(defaultPort),
 				},
-				ReadinessProbe: &corev1.Probe{
-					Handler: corev1.Handler{
-						TCPSocket: &corev1.TCPSocketAction{
-							Port: intstr.FromInt(defaultPort),
-						},
-					},
-					InitialDelaySeconds: initialDelaySeconds,
-					TimeoutSeconds:      timeoutSeconds,
+			},
+			InitialDelaySeconds: initialDelaySeconds,
+			TimeoutSeconds:      timeoutSeconds,
+			FailureThreshold:    failureThreshold,
+		},
+		ReadinessProbe: &corev1.Probe{
+			Handler: corev1.Handler{
+				TCPSocket: &corev1.TCPSocketAction{
+					Port: intstr.FromInt(defaultPort),
 				},
-				Env:       env,
-				Resources: resourceRequirements,
 			},
+			InitialDelaySeconds: initialDelaySeconds,
+			TimeoutSeconds:      timeoutSeconds,
 		},
+		Env:       env,
+		Resources: resourceRequirements,
+	}
+
+	if p.config.Pod.Container != nil {
```

Comment:
```suggestion
	if p.config.Pod != nil && p.config.Pod.Container != nil {
```

I believe this was the main reason for why the integration tests were failing.

Reply:
Oh yes! Nice catch!